### DISCUSSION TITLE
Bringing kmath into a programming language interpreter

### DISCUSSION THREAD

**Discussion by lokedhs** [mipt-npm/kmath#525] [Show and tell]
I'm working on an interpreter for a new programming language that is inspired by (and very much based on) APL: https://kapdemo.dhsdevelopments.com/

I recently replaced my custom implementation of complex numbers with that of kmath. I did this not because complex arithmetic is particularly hard, but because I have some future plans to expand the language's maths capabilities and I would prefer to be able to delegate some of this work to a third-party library instead of writing it all myself.

I'm posting this here for two reasons. The first is obviously to share information about my project so that you know where it's being used.

The second is to ask a question about the state of this project. I'm fully aware that nothing in this library can be called "stable" at this point, but I generally have no issues with that, and I'm sure the complex arithmetic API's won't change very much. But is there a plan for this project to ever become "stable" in the future, or is this a pure research project?

**Comment by SPC-code** [comment]
The answer to the "status" question is "development". The project is currently a community effort. We've started to use some parts of KMath in SPC commercial projects, but the library is currently too vast to be covered by our use-cases.  The main problem right now is documentation and test coverage. We just don't have manpower to do it properly.

So, we are looking for sponsors and contributors. But we do not plan to abandon the project.

**Comment by lokedhs** [comment]
Thank you very much for the information. I'll use it for now, since it has the potential of being precisely what I need for my project.
